src/BSON/PackedArray.c

```diff
@@ -401,6 +421,48 @@ static HashTable* php_phongo_packedarray_get_properties(phongo_compat_object_han
 	return php_phongo_packedarray_get_properties_hash(object, false, 1);
 }
 
+zval* php_phongo_packedarray_read_dimension(phongo_compat_object_handler_type* object, zval* offset, int type, zval* rv)
```

Comment:
> If implementing the interface is preferable to communicate the fact that such array access is possible, we can do so.

I think that's the right call. [ArrayObject](https://www.php.net/arrayobject) does the same in [spl_array.c](https://github.com/php/php-src/blob/php-8.2.10/ext/spl/spl_array.c#L673).

My thought is that some userland code might use an `array|ArrayAccess` type hint, which would otherwise prohibit our objects.